subprojects/gdk_rust/gdk_electrum/src/account.rs

```diff
@@ -1015,50 +1015,77 @@ pub fn create_tx(
+                    &acc_store.all_txs,
+                    &acc_store.unblinded,
+                    account.script_type,
+                ); // "policy asset" is last, in bitcoin max 1 element
+                info!("needs: {:?}", needs);
+                if needs.is_empty() {
+                    // SUCCESS tx doesn't need other inputs
+                    break;
+                }
+                let current_need = needs.pop().unwrap(); // safe to unwrap just checked it's not empty
+
+                // taking only utxos of current asset considered, filters also utxos used in this loop
+                let mut asset_utxos: Vec<&(BEOutPoint, UTXOInfo)> = utxos
+                    .iter()
+                    .filter(|(o, i)| i.asset_id() == current_need.asset && !used_utxo.contains(o))
+                    .collect();
+
+                // sort by biggest utxo, random maybe another option, but it should be deterministically random (purely random breaks send_all algorithm)
+                asset_utxos.sort_by(|a, b| (a.1).value.cmp(&(b.1).value));
+                let utxo = asset_utxos.pop().ok_or(Error::InsufficientFunds)?;
+
+                match network.id() {
+                    NetworkId::Bitcoin(_) => {
+                        // UTXO with same script must be spent together
+                        for other_utxo in utxos.iter() {
+                            if (other_utxo.1).script == (utxo.1).script {
+                                used_utxo.insert(other_utxo.0.clone());
+                                tx.add_input(other_utxo.0.clone());
+                            }
+                        }
+                    }
+                    NetworkId::Elements(_) => {
+                        // Don't spend same script together in liquid. This would allow an attacker
+                        // to cheaply send assets without value to the target, which will have to
+                        // waste fees for the extra tx inputs and (eventually) outputs.
+                        // While blinded address are required and not public knowledge,
+                        // they are still available to whom transacted with us in the past
+                        used_utxo.insert(utxo.0.clone());
+                        tx.add_input(utxo.0.clone());
                     }
                 }
             }
-            NetworkId::Elements(_) => {
-                // Don't spend same script together in liquid. This would allow an attacker
-                // to cheaply send assets without value to the target, which will have to
-                // waste fees for the extra tx inputs and (eventually) outputs.
-                // While blinded address are required and not public knowledge,
-                // they are still available to whom transacted with us in the past
-                used_utxo.insert(utxo.0.clone());
+        }
+        UtxoStrategy::Manual => {
+            let needs = tx.needs(
```

Comment:
Could this instead add the inputs first, then check that `needs()` is empty?

Reply:
Yes, that's much better, updated the last commit to do so. Thanks!